### ISSUE TITLE
Exif data missing on iOS but working on Android

### ISSUE THREAD

**Issue by zubjorge** [react-native-community/react-native-cameraroll#125]
# Bug

I am taking photos with **react-native-camera** and using this lib in combination with **piexifjs** and **react-native-fs** to set up Exif data. 

## Environment info

System:
    OS: macOS 10.15.2
    CPU: (8) x64 Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
    Memory: 2.00 GB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 13.2.0 - /usr/local/bin/node
    Yarn: 1.19.2 - /usr/local/bin/yarn
    npm: 6.13.1 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
    Android SDK:
      API Levels: 22, 23, 25, 26, 27, 28, 29
      Build Tools: 28.0.3, 29.0.2
      System Images: android-24 | Google Play Intel x86 Atom
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5791312
    Xcode: 10.1/10B61 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6
    react-native: 0.60.4 => 0.60.4
  npmGlobalPackages:
    react-native-cli: 2.0.1

Library version: 1.2.1

## Steps To Reproduce

I'm setting EXIF data with **piexif**, creating the exif object and dumping it into the original photo data. Exif data is not present on iOS but it is on Android.

```
const exifObj = {
    '0th': zeroth,
    Exif: exif,
    GPS: gps,
  }
const exifStr = piexif.dump(exifObj)
const inserted = piexif.insert(exifStr, jpegData)
const newData = await removeJpegPrefix(inserted)
```
to finally save it with this library using **CameraRoll.saveToCameraRoll**. But when I go to the gallery, the photo has no exif data on iOS, but it has exif on Android.

Describe what you expected to happen:

1. iOS photo has Exif data.

**Comment by Kuhne1** [comment]
The reason for this is that on the IOS side in the RNCCameraRollManager.m File.
The library uses, "creationRequestForAssetFromImage" if you look into the apple developer docs you will find this section. 

"A UIImage object does not contain all metadata associated with the image file it was originally loaded from (for example, Exif tags such as geographic location, camera model, and exposure parameters). To ensure such metadata is saved in the Photos library, instead use the creationRequestForAssetFromImageAtFileURL: method or the PHAssetCreationRequest class. To copy metadata from one file to another, see Image I/O."

https://developer.apple.com/documentation/photokit/phassetchangerequest/1624060-creationrequestforassetfromimage?language=objc

This is the part of the code that strips the metadata.
if ([options[@"type"] isEqualToString:@"video"]) {
        assetRequest = [PHAssetChangeRequest creationRequestForAssetFromVideoAtFileURL:inputURI];
      } else {
        assetRequest = [PHAssetChangeRequest creationRequestForAssetFromImage:inputImage];
      }

Now if you attempted the same experiment with saving a video, you would notice that the EXIF data is maintained. That is because the video method is saved via the "creationRequestForAssetFromVideoAtFileURL" which preserves metadata. 

A fix I used to preserve the metadata is to handle the image files the same as the video files.

Code to change ----
***Old*** (Within the RCT_EXPORT_METHOD(saveToCameraRoll))
        assetRequest = [PHAssetChangeRequest creationRequestForAssetFromImage:inputImage];
***New***
        assetRequest = [PHAssetChangeRequest creationRequestForAssetFromImageAtFileURL:inputURI];

Secondly, lower down in the method.
***Old*** 
  void (^loadBlock)(void) = ^void() {
 if ([options[@"type"] isEqualToString:@"video"]) {
      inputURI = request.URL;
      saveWithOptions();
    } else {
      [[self->_bridge moduleForClass:[RCTImageLoader class]] loadImageWithURLRequest:request callback:^(NSError *error, UIImage *image) {
        if (error) {
          reject(kErrorUnableToLoad, nil, error);
          return;
        }

        inputImage = image;
        saveWithOptions();
      }];
    }
}

***New***
  void (^loadBlock)(void) = ^void() {
 inputURI = request.URL;
      saveWithOptions();
};

This will mean that the library will save the photo using the URL and therefore will keep the metadata associated.

I pretext this answer, with the fact that I am not a native swift developer, this is just the fix I came up with through my research. If we have someone more familiar with swift to explain perhaps why the library choose to implement the saving on images through the "creationRequestForAssetFromImage:" It would be greatly appreciated.

**Comment by zubjorge** [comment]
Should this possible fix be included into master ?

**Comment by phyzical** [comment]
ive created a fork and a pr here https://github.com/react-native-community/react-native-cameraroll/pull/134

**Comment by bartolkaruza** [comment]
Fixed by
#134 🎉 This PR is included in version 1.4.1 🎉

The release is available on:

npm package (@latest dist-tag)
GitHub release
Your semantic-release bot 📦🚀

**Comment by ckasek** [comment]
@bartolkaruza This was re-broken due to the changes in #159.  

With 1.4.1, if I take a picture with react-native-camera and pass the URI to CameraRoll.save(), it keeps all of the exif data.  
With the latest version, the exif data is lost.
